daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/ParseErrors.scala

```diff
@@ -39,6 +40,14 @@ class ParseError(rd: Maybe[SchemaFileLocation], val loc: Maybe[DataLocation], ca
   override def toParseError = this
 }
 
+final class CharsetNotByteAlignedError(pstate: PState,
+  cause: BitsCharsetDecoderUnalignedCharDecodeException)
+  extends ParseError(
```

Comment:
Should this be a schema definition error? The only way this *should* be possible if is someone sets alignmentKind to manual but their schema isn't written correctly for manual alignment. So it's really a sign of a bad schema and maybe we don't want the chance to recover via backtracking?

Reply:
This is a tricky point. I went with Parse error because the notion is that a schema could be speculating down a branch where the alignment doesn't work out, and this error allows it to backtrack to the proper branch where it does. This is very analogous to speculating down a branch where decoding characters causes failures or delimiters aren't found properly.